src/m365/aad/commands/app/app-add.ts

```diff
@@ -441,18 +441,54 @@ class AadAppAddCommand extends GraphCommand {
     delete v2Manifest.id;
     delete v2Manifest.appId;
     delete v2Manifest.publisherDomain;
+    
     // extract secrets from the manifest. Store them in a separate variable
-    // and remove them from the manifest because we need to create them
-    // separately
-    const secrets: { name: string, expirationDate: Date }[] = this.getSecretsFromManifest(v2Manifest);
+    let secrets: { name: string, expirationDate: Date }[] = this.getSecretsFromManifest(v2Manifest);
+
+    if (args.options.withSecret) {
+      secrets = [];
+    }
+    
     // Azure Portal returns v2 manifest whereas the Graph API expects a v1.6
 
     if (args.options.apisApplication || args.options.apisDelegated) {
       // take submitted delegated / application permissions as options
       // otherwise, they will be skipped in the app update
       v2Manifest.requiredResourceAccess = appInfo.requiredResourceAccess;
     }
+
+    if (args.options.redirectUris) {
+      // take submitted redirectUris/platform as options
+      // otherwise, they will be removed from the app
+      v2Manifest.replyUrlsWithType = [];
+      
+      args.options.redirectUris.split(',').map(u => u.trim()).forEach(u => {
+        v2Manifest.replyUrlsWithType.push({ url: u, type: this.translatePlatformToType(args.options.platform!) });
+      });
+    }
+    logger.logToStderr(JSON.stringify(v2Manifest));
```

Comment:
Is it intentional that we always log the manifest to stderr? We might do it in debug mode, if at all, but logging it always seems excessive.

Reply:
I think I used this for debugging. removing it.